build/build.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+if ! which go > /dev/null; then
+	echo "error: golang needs to be installed"
+	exit 1
+fi
+
+BIN_DIR="$(pwd)/build/output/bin"
+mkdir -p "${BIN_DIR}"
+PROJECT_NAME="m3db-operator"
+REPO_PATH="github.com/m3db/m3db-operator"
+BUILD_PATH="${REPO_PATH}/cmd/${PROJECT_NAME}"
+echo "building ${PROJECT_NAME} ..."
```

Comment:
I noticed there's no `go clean` in this file, or `make clean` (with `go clean` and `rm --whatever`) in the Makefile, maybe that would be useful?

Reply:
Sure